packages/jsx-email/src/render/process.ts

```diff
@@ -8,6 +8,8 @@ import type { ProcessOptions } from '../types';
 
 export { prettyHtml };
 
+export const jsxEmailTags = ['jsx-email-cond'];
```

Comment:
Why don't we make this 
```tsx
export const jsxEmailTags = ['Jsx-email-cond'];
```

so we can avoid the ts-ignore above

Reply:
https://react.dev/reference/react-dom/components#custom-html-elements

We don't define any behavior for it (following https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements) because we don't need any behavior at all; we're removing it from the output during the process stage of render.

Then, consider the info in this issue https://github.com/microsoft/TypeScript/issues/4648, and the last comment:

> It's part of jsx - lowercased names are considered intrinsics, while uppercased names are custom components. They get called differently, too. Intrinsics are just a string passed into createElement, while custom components pass in their constructor.

We want it to be treated as an intrinsic so it's called the same way as other intrinsics, and treated as a custom element, and not a custom component, as stated in the react docs. That's semantically correct, and works as intended, as can be seen here (inspect to see tags): https://reactplayground.vercel.app/#N4IglgdgJgpgHgOgFYGcQC5wFsAOB7AJwBcACAQRxxIDMC8sSByBAegp2RTkYB0IxchUgCUYAQwDGpWvSYFxU3v0HESwEhPliiMYXjykAvjToNGWqQFoo9FhIA2YGBCJK+EvBBSkPLsZBgCEgBeEhsJAFcsZyIEAHMYIgBRexholwAhAE8ASSgACnN9VwBKd09vEjoDEI0tHT0DfN8ifwhAsohq2PloQPy+EhJRSVjNcR0UtJiBiCGhkakEAGUiAjApAFk8WAAaQfmICPt7fbn5xbH6mCn0onz2TqHOkpBdkHZOOAxsfFVL3ZqEgRFAwVbaGAkYwyMwWVx8AR-UjeLKpKAmWSMFFoyweQTtFwoNwQXyVAAiYggCToIOEYgA7gAJIhYey1fLqAAWLLZoR4IH5UJKIQAfCRZkMADyoOCWNL+ey4zzoqCU6l4EH2LLLRI5CDtAiMgAqmwAMsFgOoAPpW7ms9AkO1swzGFgivglADcfB9JIqpHY7OFwTFwAOpNIAG0PBEXIDQUQAMIalwAXVqILBrR0+QADJ0DvIiBECHMJfNJe7zlKKVTAhqUHSmTzHTyLQADA7zEiSgCElkskbA1BIvawKDwqZFfYHVe73clnIAjCLOTATnhJSxl3P5zPLH3I84oMPU7OuyR267d1KMhEiERPCRPInHBIANYW-LBsUJ5Ox+4YxcEgAGoSCXEpDBvCsUBwSkqjwVJgn5AQ4kFMR1jESx7DEAAjddkJAOE5SwPAkDAfkRUAbLJAHg-rdYMpUMgKIQwLy3O8H08aCpTdA5OlYiByi8UgOMfOZQmxGAoAQXD7zEztzmoTwiEsFAwAALxgB0ACZ5Cwb0IAUvh4CRMIYGoMRjgDSgQEMIA

So in all, we want it as it currently is. telling TS to ignore it is the preferred escape hatch.

Reply:
Thanks for the explanation. I understand the approach a lot more now